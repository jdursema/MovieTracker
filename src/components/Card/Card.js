import React from 'react';
import './Card.css';
import PropTypes from 'prop-types';

const Card = props => {
  const movieObj = {
    title: props.title,
    poster_path: props.posterPath,
    movie_id: props.id,
    release_date: props.releaseDate,
    vote_average: props.voteAvg,
    overview: props.overview
  };

  const handleCardClick = () => {

    if (!props.user.id) {
      props.history.push('/login');
    } else if (props.user.id && !props.isFav) {
      props.handleAddFav(props.user.id, movieObj);
      console.log('handleAdd', props.isFav)
    } else if (props.user.id && props.isFav){
      props.handleRemoveFav(props.user.id, props.id);
      console.log('handleRemoveFav', props.isFav)
    }
  };

  return (
    <div className="Card">
      <img
        className="poster"
        src={`https://image.tmdb.org/t/p/w500${props.posterPath}`}
        alt={`Poster for ${props.title}`}
      />
      <div className="overlay">
        <img
        className="backdrop"
        src={`https://image.tmdb.org/t/p/w500${props.backdropPath}`}
        alt={`Back drop for ${props.title}`}
        />
        <button
          onClick={handleCardClick}>
          Fav
        </button>
        <h3>{props.title}</h3>
        <h4>Released: {props.releaseDate}</h4>
        <h4>Rating: {props.voteAvg}</h4>
        <h4>{props.overview}</h4>
      </div>
    </div>
  );
};

export default Card;


Card.propTypes = {
  user: PropTypes.object,
  title: PropTypes.string,
  posterPath: PropTypes.string,
  handleAddFav: PropTypes.func,
  handleRemoveFav: PropTypes.func,
  id: PropTypes.number,
  releaseDate: PropTypes.string,
  voteAvg: PropTypes.number,
  overview: PropTypes.string,
  history: PropTypes.object
};