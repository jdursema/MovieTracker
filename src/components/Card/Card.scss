.Card {
	width: 425px;
	height: 600px;
	margin: 15px;
	position: relative;
	flex: 0 0 auto;
	&:hover .overlay {
		background-color: #222020;
		color: #FFF;
		opacity: 1;
	}
	
	.poster {
		width: 425px;
		height: 600px;
		opacity: 1;
		position: absolute;
		transition: opacity .5s ease;		
		&:hover {
			opacity: 0;
			z-index: 0;
		}
	}

	.overlay {
		opacity: 0;
		height: 100%;
		
		.backdrop {
			width: 100%;		
		}
		
		h3, h4 {
			line-height: 1.3;
			margin: 8px 12px;
		}
		
		h3 {
			font-size: 24px;
		}
		
		button {
			position: absolute;
			top: 100px;
			left: 200px;
			color: #FFF;
		}
  }
}