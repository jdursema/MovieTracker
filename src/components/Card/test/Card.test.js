import React from 'react';
import { shallow } from 'enzyme';
import Card from '../Card.js';

describe ('Card', () => {
  let renderedCard;
  let mockHandleAddFav;
  let mockHandleRemoveFav;

  beforeEach(()=>{
    mockHandleAddFav = jest.fn()
    mockHandleRemoveFav = jest.fn()
    renderedCard = shallow(<Card 
      handleAddFav= {mockHandleAddFav} handleRemoveFav={mockHandleRemoveFav}
      user={{id: 3}}
      isFav= {false}/>) 
  })
  it('should match the snapshot', () => {
  
    expect(renderedCard).toMatchSnapshot()
  })

  it.skip('should call handleCardClick when the button is clicked', () => {


  handleCardClick()

  expect(handleAddFav).toHaveBeenCalled()
  });


})

