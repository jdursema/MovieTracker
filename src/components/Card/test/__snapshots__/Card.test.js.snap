// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    handleAddFav={[Function]}
    handleRemoveFav={[Function]}
    isFav={false}
    user={
        Object {
            "id": 3,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Poster for undefined"
          className="poster"
          src="https://image.tmdb.org/t/p/w500undefined"
/>,
        <div
          className="overlay"
>
          <img
                    alt="Back drop for undefined"
                    className="backdrop"
                    src="https://image.tmdb.org/t/p/w500undefined"
          />
          <button
                    className="card-fav"
                    onClick={[Function]}
          />
          <h3 />
          <h4>
                    Released: 
          </h4>
          <h4>
                    Rating: 
          </h4>
          <h4 />
</div>,
      ],
      "className": "Card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Poster for undefined",
          "className": "poster",
          "src": "https://image.tmdb.org/t/p/w500undefined",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Back drop for undefined"
              className="backdrop"
              src="https://image.tmdb.org/t/p/w500undefined"
/>,
            <button
              className="card-fav"
              onClick={[Function]}
/>,
            <h3 />,
            <h4>
              Released: 
</h4>,
            <h4>
              Rating: 
</h4>,
            <h4 />,
          ],
          "className": "overlay",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Back drop for undefined",
              "className": "backdrop",
              "src": "https://image.tmdb.org/t/p/w500undefined",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "card-fav",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Released: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Released: ",
              undefined,
            ],
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Rating: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Rating: ",
              undefined,
            ],
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h4",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Poster for undefined"
            className="poster"
            src="https://image.tmdb.org/t/p/w500undefined"
/>,
          <div
            className="overlay"
>
            <img
                        alt="Back drop for undefined"
                        className="backdrop"
                        src="https://image.tmdb.org/t/p/w500undefined"
            />
            <button
                        className="card-fav"
                        onClick={[Function]}
            />
            <h3 />
            <h4>
                        Released: 
            </h4>
            <h4>
                        Rating: 
            </h4>
            <h4 />
</div>,
        ],
        "className": "Card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Poster for undefined",
            "className": "poster",
            "src": "https://image.tmdb.org/t/p/w500undefined",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="Back drop for undefined"
                className="backdrop"
                src="https://image.tmdb.org/t/p/w500undefined"
/>,
              <button
                className="card-fav"
                onClick={[Function]}
/>,
              <h3 />,
              <h4>
                Released: 
</h4>,
              <h4>
                Rating: 
</h4>,
              <h4 />,
            ],
            "className": "overlay",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Back drop for undefined",
                "className": "backdrop",
                "src": "https://image.tmdb.org/t/p/w500undefined",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "card-fav",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Released: ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Released: ",
                undefined,
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Rating: ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Rating: ",
                undefined,
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h4",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
